### DISCUSSION TITLE
MQTT, I have a couple of doubts that are running out for me.

### DISCUSSION THREAD

**Discussion by jesusmorales-mape** [Mair/esp32-course#88] [Course Qeustions] [answered]
i have a problem, i can´t connect to MQTT broker, no user, no password, secure connection (TLS) but whithout certificaction, 

struct esp_mqtt_client_config_t mqttConfig = { .uri = "mqtt://devices.factorylive.io:8883" }; 

i have this errors E (437159) MQTT_CLIENT: mqtt_message_receive: transport_read() error: errno=128 I (437159) MQTT: MQTT_EVENT_ERROR E (437159) MQTT_CLIENT: esp_mqtt_connect: mqtt_message_receive() returned -1 E (437169) MQTT_CLIENT: MQTT connect failed I (437179) MQTT: MQTT_EVENT_DISCONNECTED I (452179) MQTT: Other event id:7 

I understand that I'm missing something in the configuration, but I don't know what it is :(
what could I be missing? 

But to broker hivemq.com:1883, I can to publish data this configuration.
struct esp_mqtt_client_config_t mqttConfig = { .uri = "mqtt://broker.hivemq.com:1883" };

**Comment by Mair** [comment]
@jesusmorales-mape  I tried to connect to that mqtt endpoint with a regular mqtt client (MQTT Explorer) and I was unable to connect.
Are you sure you have the endpoint and port correct?

**Reply by jesusmorales-mape** [reply]
I already solved the problem, I was missing the ssl certificate, add it and that's it. thanks

**Comment by Mair** [comment, marked as answer]
> I already solved the problem, I was missing the ssl certificate, add it and that's it. thanks


well done
